apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/account/service/AliasServiceImplTest.java

```diff
@@ -0,0 +1,341 @@
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author silaev-firstbridge on 3/19/2020
+ */
+@ExtendWith(MockitoExtension.class)
+class AliasServiceImplTest {
+    @Mock
+    private AliasTable aliasTable;
+    @Mock
+    private AliasOfferTable offerTable;
+    @Mock
+    private Blockchain blockchain;
+    @InjectMocks
+    @Spy
+    private AliasServiceImpl aliasService;
+
+    @Test
+    void shouldGetCount() {
+        //GIVEN
+
+        //WHEN
+        aliasService.getCount();
+
+        //THEN
+        verify(aliasTable, times(1)).getCount();
+    }
+
+    @Test
+    void shouldGetAccountAliasCount() {
+        //GIVEN
+        final long accountId = 7821792282123976600L;
+
+        //WHEN
+        aliasService.getAccountAliasCount(accountId);
+
+        //THEN
+        verify(aliasTable, times(1)).getCount(any(DbClause.LongClause.class));
+    }
+
+    @Test
+    void shouldGetAliasesByOwner() {
+        //GIVEN
+        final long accountId = 7821792282123976600L;
+        final int from = 1;
+        final int to = 2;
+
+        //WHEN
+        aliasService.getAliasesByOwner(accountId, from, to);
+
+        //THEN
+        verify(aliasTable, times(1)).getManyBy(any(DbClause.LongClause.class), eq(from), eq(to));
+    }
+
+    @Test
+    void shouldGetAliasByString() {
+        //GIVEN
+        final String aliasName = "AS1561618989348";
+
+        //WHEN
+        aliasService.getAlias(aliasName);
+
+        //THEN
+        verify(aliasTable, times(1)).getBy(any(DbClause.StringClause.class));
+    }
+
+    @Test
+    void shouldGetAliasesLike() {
+        //GIVEN
+        final String aliasName = "AS1561618989348";
+        final int from = 1;
+        final int to = 2;
+
+        //WHEN
+        aliasService.getAliasesLike(aliasName, from, to);
+
+        //THEN
+        verify(aliasTable, times(1))
+            .getManyBy(any(DbClause.LikeClause.class), eq(from), eq(to));
+    }
+
+    @Test
+    void shouldGetAliasById() {
+        //GIVEN
+        final long id = 1815781403495190811L;
+
+        //WHEN
+        aliasService.getAlias(id);
+
+        //THEN
+        verify(aliasTable, times(1)).getAlias(id);
+    }
+
+    @Test
+    void shouldDeleteOnlyAlias() {
+        //GIVEN
+        final String aliasName = "AS1561618989348";
+        final Alias alias = mock(Alias.class);
+        final int height = 1445;
+        when(aliasTable.getBy(any(DbClause.StringClause.class))).thenReturn(alias);
+        when(blockchain.getHeight()).thenReturn(height);
+
+        //WHEN
+        aliasService.deleteAlias(aliasName);
+
+        //THEN
+        verify(offerTable, times(0)).deleteAtHeight(null, height);
+        verify(aliasTable, times(1)).deleteAtHeight(alias, height);
+    }
+
+    @Test
+    void shouldDeleteOnlyAliasAndOffer() {
```

Comment:
alias and offer represent complete set of entities to delete, no need 'only'

Reply:
done